src/js/providers/default.ts

```diff
@@ -1,9 +1,68 @@
 import { PLAYER_STATE, MEDIA_TYPE } from 'events/events';
+import { GenericObject, SourceObj, SeekRange } from 'types/generic.type';
+
+const noop: () => void = function(): void { /* noop */ };
+const returnFalse: () => boolean = (() => false);
+const getNameResult: { name: string } = { name: 'default' };
+const returnName: () => { name: string } = (() => getNameResult);
+interface DefaultProvider {
+    state?: string;
+    supports: () => boolean;
+    play: () => void;
+    pause: () => void;
+    preload: () => void;
```

Comment:
item arg

Reply:
oof this is going to take a lot of refactoring of the noops below. gonna dive back in

Reply:
Yeah I hesitated to give you this feedback. On one hand the purpose of this module is to "plug holes" in provider implementations by providing a base no-op implementation. On the other, more ideally, it should be a base class and interface for providers to implement.

Reply:
Came up with a better solution using extends to retain type safety but will be typing these funcs as implemented in this pr as well